randovania/server/server_app.py

```diff
@@ -406,38 +427,66 @@ async def server_app(request: fastapi.Request) -> ServerApp:
 """The `ServerApp` handling this request."""
 
 
-def get_user(needs_admin: bool) -> AsyncCallable[[ServerAppDep, fastapi.Request], User]:
+async def get_user(
+    sa: ServerAppDep,
+    request: fastapi.Request,
+    x_randovania_session: Annotated[str | None, fastapi.Header()] = None,
+) -> User:
     """
-    Gets the User associated with the Request, and optionally checks admin permissions.
+    Gets the User associated with the Request.
     Used for dependency injection.
     """
 
-    async def handler(sa: ServerAppDep, request: fastapi.Request) -> User:
-        try:
-            user: User
-            if not sa.app.debug:
-                token = request.session.get("discord_oauth_token")
-                discord_user = await sa.discord.user(token)
-                user = User.get(discord_id=int(discord_user.id))
-                if user is None or (needs_admin and not user.admin):
-                    raise fastapi.HTTPException(status_code=403, detail="User not authorized")
-            else:
-                user = User.get()
+    user_id: int | None = request.session.get("user_id")
+    discord_token = request.session.get("discord_oauth_token")
 
-            return user
+    # TODO: check for user tokens
+    # maybe make sure tokens have a prefix so we know it's token vs encrypted session?
```

Comment:
could you clarify this TODO? is this related to a change in behavior, or did the old function have this deficiency as well?